src/test-helpers/typeFactory.ts

```diff
@@ -37,13 +58,29 @@ export class TypeFactory {
 		this.#registry = this.#api.registry;
 	}
 
-	storageKey = (index: number): StorageKey => {
-		const key = new StorageKey(
-			this.#registry,
-			this.#api.query.crowdloan.funds.key(this.paraIndex(index))
-		);
-
-		return key.setMeta(this.#api.query.crowdloan.funds.creator.meta);
+	/**
+	 * @param index
+	 * The id to assign the key to.
+	 * @param indexType
+	 * The InterfaceType that will be used to create the index into its new appropriate index type
+	 * @param storageEntry
+	 * Used primarily on QueryableStorageEntry (ie: api.query) within the polkadot api library.
+	 * Contains necessary key value pairs to retrieve specific information from a query
+	 * such as `at`, `entriesAt`, `entries` etc..
+	 *
+	 * Some Parameter Examples:
+	 * 1. this.api.query.crowdloans.funds
+	 * 2. this.api.query.slots.leases
+	 */
+	storageKey = (
```

Comment:
Is there any reason these are not declared as `function storagekey(..) { .. }`? If not I prefer `function` just to signal that we don't expect it to have any issues with binding `this`.

Reply:
Fixed, yea I agree here.